tests/foreman/ui/test_host.py

```diff
@@ -1,14 +1,278 @@
+            organization=cls.org).create()
+        cls.content_view.repository = [cls.repo]
+        cls.content_view = cls.content_view.update(['repository'])
+        cls.content_view.publish()
+        cls.content_view = cls.content_view.read()
+        promote(cls.content_view.version[0], cls.lc_env.id)
+
+        # Search for puppet environment and associate location
+        cls.environment = entities.Environment().search(
+            query={'organization_id': [cls.org.id]}
+        )[0]
+        cls.environment.location = [cls.loc]
+        cls.environment = cls.environment.update(['location'])
+
+        # Search for SmartProxy, and associate location
+        cls.proxy = entities.SmartProxy().search(
+            query={
+                u'search': u'name={0}'.format(
+                    conf.properties['main.server.hostname'])
+            }
+        )[0]
+        cls.proxy.location = [cls.loc]
+        cls.proxy = cls.proxy.update(['location'])
+
+        # Search for domain and associate org, location
+        cls.domain = entities.Domain().search(
+            query={
+                u'search': u'name="usersys.redhat.com"'
+            }
+        )[0]
+        cls.domain_name = cls.domain.name
+        cls.domain.location = [cls.loc]
+        cls.domain.organization = [cls.org]
+        cls.domain.dns = cls.proxy
+        cls.domain = cls.domain.update(['dns', 'location', 'organization'])
+
+        # Search if subnet is defined with given network.
+        # If so, just update its relevant fields otherwise,
+        # Create new subnet
+        network = '192.168.100.0'
+        cls.subnet = entities.Subnet().search(
+            query={u'search': u'network={0}'.format(network)})
+        if (len(cls.subnet) == 1):
+            cls.subnet = cls.subnet[0]
+            cls.subnet.domain = [cls.domain]
+            cls.subnet.location = [cls.loc]
+            cls.subnet.organization = [cls.org]
+            cls.subnet.dns = cls.proxy
+            cls.subnet.dhcp = cls.proxy
+            cls.subnet.tftp = cls.proxy
+            cls.subnet.discovery = cls.proxy
+            cls.subnet = cls.subnet.update(
+                ['domain',
+                 'discovery',
+                 'dhcp',
+                 'dns',
+                 'location',
+                 'organization',
+                 'tftp']
+            )
+        else:
+            # Create new subnet
+            cls.subnet = entities.Subnet(
+                name=gen_string('alpha'),
+                network=network,
+                mask=u'255.255.255.0',
+                domain=[cls.domain],
+                location=[cls.loc],
+                organization=[cls.org],
+                dns=cls.proxy,
+                dhcp=cls.proxy,
+                tftp=cls.proxy,
+                discovery=cls.proxy
+            ).create()
+
+        # Create Libvirt Compute-resource
+        cls.computeresource = entities.LibvirtComputeResource(
```

Comment:
Perhaps you should search if a compute resource already exists, and if so, use it, otherwise, create it? I ask this because if there is one already, creating it would raise an error (uniqueness of URL or something like that)